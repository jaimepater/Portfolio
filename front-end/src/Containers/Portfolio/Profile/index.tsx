import * as React from 'react';
import { observer } from 'mobx-react';
import { useEffect } from 'react';
import useStores from '../../../Commons/Hooks/Stores';
import Profile from '../../../Components/Profile/Profile';

const ProfilePortfolio = observer(() => {
  const { profileStore } = useStores();
  useEffect(() => {
    profileStore.getProfileData();
  }, [profileStore]);

  const { fullName, profiles } = profileStore;
  const profile = {
    name: fullName,
    showFlag: true,
    list: profiles,
  };
  return <Profile {...profile} />;
});

export default ProfilePortfolio;
