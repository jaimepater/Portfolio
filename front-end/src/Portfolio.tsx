import React, { Suspense } from 'react';

import './App.css';
import { ThemeProvider } from 'styled-components';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
// eslint-disable-next-line import/no-extraneous-dependencies
import { Helmet } from 'react-helmet';
import { GlobalStyle, Theme } from './Theme/Theme';
import Header from './Components/Header';
import ContentGradient from './Components/ContentGradient';
import Footer from './Components/Footer';
import { getRoutes, RoutesPageTypes } from './Commons/Utils/getRoutes';
import logo from './assets/Images/logos/logo.png';

const PortfolioApp = () => (
  <ThemeProvider theme={Theme}>
    <div className="App">
      <Helmet>
        <title>Jaime Paternina Portfolio</title>
        <meta
          name="description"
          content="Portfolio of Jaime Paternina showcasing skills, projects, and experience."
        />

        <meta property="og:title" content="Jaime Paternina's Portfolio" />
        <meta
          property="og:description"
          content="Portfolio of Jaime Paternina showcasing skills, projects, and experience."
        />
        <meta property="og:image" content={logo} />
        <meta property="og:url" content="https://jaime-portfolio.com/" />
        <meta property="og:type" content="website" />
      </Helmet>
      <Router>
        <Suspense fallback={<div>Loading...</div>}>
          <Header />
          <ContentGradient>
            <Switch>
              {getRoutes(RoutesPageTypes.PORTFOLIO).map(route => (
                <Route
                  key={route.path}
                  exact={route.exact}
                  path={route.path}
                  component={route.component}
                />
              ))}
            </Switch>
            <Footer />
          </ContentGradient>
        </Suspense>
      </Router>
      <GlobalStyle />
    </div>
  </ThemeProvider>
);

export default PortfolioApp;
