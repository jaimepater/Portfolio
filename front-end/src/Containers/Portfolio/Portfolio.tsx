import * as React from 'react';
import Profile from './Profile/Profile';
import Experience from './Experience/Experience';
import Skills from './Skills/Skills';
import Projects from './Projects/Projects';
import Hobbies from './Hobbies/Hobbies';

const Portfolio = () => (
  <>
    <Profile />
    <Experience />
    <Skills />
    <Projects />
    <Hobbies />
  </>
);

export default Portfolio;
