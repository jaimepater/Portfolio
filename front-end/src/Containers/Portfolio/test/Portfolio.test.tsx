import React from "react";
import { shallow } from 'enzyme';
import Portfolio from "../Portfolio";


describe('Portfolio', () => {
    it('Renders Header', () => {
        const shallowComponent = shallow(<Portfolio  />);
        expect(shallowComponent).toMatchSnapshot();
    });
});