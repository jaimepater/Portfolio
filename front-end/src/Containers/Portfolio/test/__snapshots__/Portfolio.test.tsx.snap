// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portfolio Renders Header 1`] = `<Profile />`;
